### PR TITLE
Add props to Autocomplete + handle disabled state on Textfield

### PR DESCRIPTION
**Pull request by Zarthia**

### CODE REVIEW

File: src/textField/index.tsx [hunk 1]
```diff
@@ -184,7 +185,7 @@ export default class TextField extends PureComponent <TextFieldProps, TextFieldS
             />
           }
           {
-            !isTextArea &&
+            shouldDisplayButton &&
```
Comment (Tuxosaurus) [inline src/textField/index.tsx]: Question about the existing: why do we always show a button component - albeit an empty one - on all inputs?

Comment (Zarthia) [reply src/textField/index.tsx]: That is a good question, that's the way it is right now

File: src/autoComplete/index.tsx [hunk 2]
```diff
@@ -90,6 +96,12 @@ export default class AutoComplete extends Component<AutoCompleteProps, AutoCompl
     defaultValue: '',
     showList: true,
     valid: false,
+    autoCorrect: 'off',
```
Comment (loick) [inline src/autoComplete/index.tsx]: Wouldn't it be better to directly extend the Textfield props? Or are there some props you don't want to have

File: src/autoComplete/index.tsx [hunk 3]
```diff
@@ -219,11 +231,15 @@ export default class AutoComplete extends Component<AutoCompleteProps, AutoCompl
           defaultValue={String(this.state.query)}
           addon={this.props.inputAddon}
           inputRef={this.inputRef}
-          autoCorrect="off"
-          autoComplete="off"
+          autoCorrect={this.props.autoCorrect}
+          autoComplete={this.props.autoComplete}
```
Comment (loick) [inline src/autoComplete/index.tsx]: autoComplete should always be off IMO

Comment (Zarthia) [reply src/autoComplete/index.tsx]: On the Autocomplete, I agree. Is that a problem @dagatsoin ?

Comment (dagatsoin) [reply src/autoComplete/index.tsx]: No.

File: src/autoComplete/index.tsx (hunk 2 above)
Comment (Zarthia) [reply src/autoComplete/index.tsx]: Do you mean those?
```
export interface CommonFormFields {
  name: string,
  id?: string,
  type?: 'text' | 'password' | 'email' | 'number' | 'search',
  placeholder?: string,
  maxLength?: number,
  autoCorrect?: 'on' | 'off',
  autoComplete?: 'on' | 'off',
  disabled?: boolean,
  readOnly?: boolean,
  autoFocus?: boolean,
  required?: boolean,
  onFocus?: (event:
    React.FocusEvent<HTMLTextAreaElement> | React.FocusEvent<HTMLInputElement>) => void,
  onBlur?: (event:
    React.FocusEvent<HTMLTextAreaElement> | React.FocusEvent<HTMLInputElement>) => void,
}
```

**Pull request merged**